packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -133,6 +141,102 @@ export class ApexDebug extends DebugSession {
     this.sendResponse(response);
   }
 
+  protected async setBreakPointsRequest(
+    response: DebugProtocol.SetBreakpointsResponse,
+    args: DebugProtocol.SetBreakpointsArguments
+  ): Promise<void> {
+    if (args.source && args.source.path) {
+      const processedBreakpoints: DebugProtocol.Breakpoint[] = [];
+      const uriPath = this.convertClientPathToDebugger(args.source.path);
+
+      try {
+        const linesToSetBreakpoint = await this.myBreakpointService.reconcileBreakpoints(
+          this.sfdxProject,
+          this.mySessionService.getSessionId(),
+          uriPath,
+          args.lines
+        );
+
+        for (const clientLine of linesToSetBreakpoint) {
+          const serverLine = this.convertClientLineToDebugger(clientLine);
+          const typeref = this.myBreakpointService.getTyperefFor(
+            uriPath,
+            serverLine
+          );
+          if (typeref && typeref.length > 0) {
+            const cmdOutput = await this.myBreakpointService.createLineBreakpoint(
+              this.sfdxProject,
+              this.mySessionService.getSessionId(),
+              typeref,
+              serverLine
+            );
+            this.myBreakpointService.cacheLiveBreakpoint(
+              uriPath,
+              clientLine,
+              cmdOutput.getId()
+            );
+            processedBreakpoints.push(
+              <DebugProtocol.Breakpoint>{
+                verified: true,
+                source: args.source,
+                line: clientLine
+              }
+            );
+          } else {
+            processedBreakpoints.push(
+              <DebugProtocol.Breakpoint>{
+                verified: false,
+                source: args.source,
+                line: clientLine
+              }
+            );
+          }
+        }
+
+        response.success = true;
+        response.body = {
+          breakpoints: processedBreakpoints
+        };
+      } catch (error) {
+        this.tryToParseSfdxError(response, error);
+      }
+    }
+
+    this.sendResponse(response);
+  }
+
+  protected customRequest(
+    command: string,
+    response: DebugProtocol.Response,
+    args: any
+  ): void {
+    switch (command) {
+      case 'lineBreakpointInfo':
```

Comment:
Apex debugger adapter receives the line breakpoint info from vscode-apex-debugger.

Reply:
Go with check the apex language server before starting a debugger session.
Since it is not likely that the language server is going to be unavailable, we can rely on it as a precondition for starting the debugger.

The only hard requirement that could prevent the language server from starting is when the user doesn't have Java properly installed on the machine.